custom_attributes.go

```diff
@@ -0,0 +1,125 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// CustomeAttributesService handles communication with the
+// group, project and user custom attributes related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/custom_attributes.html
+type CustomeAttributesService struct {
+	client *Client
+}
+
+// CustomAttributes struct is used to unmarshal response to api calls
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/custom_attributes.html
+type CustomAttributes struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+// CustomAttributeOptions struct contains all options required to call the methods below
+// agianst the gitlab api for customattributes
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/custom_attributes.html
+type CustomAttributeOptions struct {
+	CustomeAttributeResourceID interface{}
+	CustomeAttributeResource   string
+	CA                         CustomAttributes
+}
+
+// ListCustomAttributes lists the custom attributes of the specified resource
+//
+// GitLab API docs:  https://docs.gitlab.com/ee/api/custom_attributes.html#list-custom-attributes
+func (s *CustomeAttributesService) ListCustomAttributes(opt CustomAttributeOptions, options ...OptionFunc) ([]*CustomAttributes, *Response, error) {
+	resourceID, err := parseID(opt.CustomeAttributeResourceID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	u := fmt.Sprintf("%s/%s/custom_attributes", opt.CustomeAttributeResource, url.QueryEscape(resourceID))
```

Comment:
Please see some of the other services for comparison... The default approach is to have the parts that make up the URL, be function parameters. And all additional fields should be in the options structs.

So for this function that should be:

```
func (s *CustomAttributesService) ListCustomAttributes(resource int, options ...OptionFunc) ([]*CustomAttribute, *Response, error) {
```

Additionally you could have a look at the `search` service: https://github.com/xanzy/go-gitlab/blob/master/search.go As you can see it uses fully named functions that then call a private function to execute the actual logic. So in this service it could be something like this:

```
func (s *CustomAttributesService) ListCustomUserAttributes(user int, options ...OptionFunc) ([]*CustomAttribute, *Response, error) {
  return s.listCustomAttributes("users", user, options...)
}
```

And

```
func (s *CustomAttributesService) ListCustomGroupAttributes(group int, options ...OptionFunc) ([]*CustomAttribute, *Response, error) {
  return s.listCustomAttributes("groups", group, options...)
}
```

Which then call a private function like:

```
func (s *CustomAttributesService) listCustomAttributes(resource string, id int, options ...OptionFunc) ([]*CustomAttribute, *Response, error) {
  u := fmt.Sprintf("%s/%d/custom_attributes", resource, id)
  req, err := s.client.NewRequest("GET", u, nil, options)
  if err != nil {
    return nil, nil, err
  }
  
  var ca []*CustomAttribute
  resp, err := s.client.Do(req, &ca)
  if err != nil {
    return nil, resp, err
  }

  return ca, resp, err
}
```

This provides a much better UX and is more consistent with the other services in this package. Of course this comment goes for all the functions of this service...